hydroshare/settings.py

```diff
@@ -854,6 +854,9 @@ def _(s): return s # noqa
 SESSION_COOKIE_SECURE = USE_SECURITY
 CSRF_COOKIE_SECURE = USE_SECURITY
 
+# TODO: ideally we can avoid setting this to False
+SESSION_COOKIE_HTTPONLY = False
```

Comment:
@sblack-usu this is the part I don't like, revealing our django session to JS so that we can send it with requests from Uppy to Companion

Maybe there's a better way to do this with a JWT?

Reply:
I think you are onto something. We might be able to do something with protocol mappers. I'll need to better understand how this is used and maybe I can recommend something. This is requests coming from the client (browser) to interact with the uppy server and the uppy server needs the django session to authenticate with django server?

Reply:
Can the cookie be available for both hydroshare and companion?

Reply:
sharing cookies across subdomains is easy enough. https://stackoverflow.com/a/23086139

Reply:
currently we have companion deployed under `cuahsi.org` -- different domain.